Java chaincode

joshua-roberts opened the pull request:
This PR adds both the Authorization and Asset chaincode functions in Java. There is also an e2e test suite that uses a sample Fabric network to test transient input and object serialization and a complete e2e scenario invoking every function at least once.

JustKuzya reviewed (commented):
I believe that **asset/libs/** path should be in the root .gitignore as you don't know when licenses of the jars may change and we(Feds) don't want to distribute "commercial or not-free-for-all" licensed software from our repo. All we should provide is dependency management reference file(s)for Maven/Gradle, or whatever you may use. This way whoever uses the code can build dependency tree and deal with the license changes independent from the our implementation.

joshua-roberts commented:
I removed the asset/libs and authorization/libs directories. They are built when running `make build-asset` from `/asset` and `make build-auth` from `/authorization`. I also added gradle wrapper files for version 5.6.2 which is compatible with Java 11.

iMichaela reviewed (approved):
LGTM
